ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/CompactorTest.java

```diff
@@ -102,6 +102,11 @@ protected CompactorTest() throws Exception {
     tmpdir.deleteOnExit();
   }
 
+  protected void deleteTableDir(String tableName) throws IOException {
```

Comment:
Just doing a org.apache.commons.io.FileUtils.deleteDirectory(tmpdir) before tmpdir.mkdir() would be a simple solution, as all files are being created under that dir. With that the individual deleteTableDir calls won't be needed.
Also, the path could be made unique by using  Files.createTempDirectory, so that concurrent running of tests won't step on each other.